engines/order_management/app/services/order_management/order/stripe_sca_payment_authorize.rb

```diff
@@ -15,7 +15,9 @@ def call!(redirect_url = full_order_path(@order))
 
         @payment.authorize!(redirect_url)
 
-        @order.errors.add(:base, I18n.t('authorization_failure')) unless @payment.pending?
+        unless @payment.pending? || @payment.requires_authorization?
+          @order.errors.add(:base, I18n.t('authorization_failure'))
+        end
```

Comment:
Because payments requiring auth now move to `requires_authorization` instead of `pending`, this change is necessary to preserve the current behavior.

Reply:
Is this needed? Won't line 10 already find only payments that are in `pending` state (which means they're not in `requires_authorization` state)?

Reply:
Line 16 (`@payment.authorize!(redirect_url)`) should either change the state to `requires_authorization` or leave it in `pending`. If there's a problem with the card (insufficient funds for example) it will move it to `failed` and we want to add this error.

Reply:
I see "@payment.pending? || @payment.requires_authorization?" appearing twice in this PR, looks like a meaningful condition. I'd make a method in the payment object and explain what it is in the method name or method docs.